overlord/devicestate/devicemgr.go

```diff
@@ -1425,6 +1425,73 @@ func (m *DeviceManager) ensurePostFactoryReset() error {
 	return os.Remove(factoryResetMarker)
 }
 
+func (m *DeviceManager) hasUserExpired(user *auth.UserState) bool {
+	// If the user has no expiration date, then Expiration should not
+	// be set, and contain the default value.
+	if user.Expiration.IsZero() {
+		return false
+	}
+
+	return user.Expiration.Before(time.Now())
+}
+
+var osutilDelUser = osutil.DelUser
```

Comment:
nitpick: I'd suggest putting this near the top of the file

Reply:
Good point!